### PR TITLE
Don't add empty item to list in OgMembershipReferenceItemList::get

### PR DESCRIPTION
**Pull request by damiankloip**
I will create a core issue for this too as the `ItemList` class suffers the same problem.

Doing the following:

``` php
$group_content->og_group_ref->count() // 0 - correct
$group_content->og_group_ref->value // NULL - correct
$group_content->og_group_ref->count() // 1 - Incorrect, should still be 0, we have no items!
```

### CODE REVIEW

File: src/Plugin/Field/FieldType/OgMembershipReferenceItemList.php [hunk 1]
```diff
@@ -51,9 +51,10 @@ public function get($index) {
       $this->fetched = TRUE;
     }
 
-    // Automatically create the first item for computed fields.
+    // Just return an empty item if the first item is requested and the list is
```
Comment (dawehner) [inline src/Plugin/Field/FieldType/OgMembershipReferenceItemList.php:54]: what about adding a comment that count() would change if we store the result?

Comment (damiankloip) [inline src/Plugin/Field/FieldType/OgMembershipReferenceItemList.php:54]: Not a bad idea.

**Comment by amitaibu** [comment]
Can you add a test for this please?

**Comment by damiankloip** [comment]
Mhhhhh. The membership item currently has no test coverage. How about a follow up solely for test coverage? :)

**Comment by amitaibu** [comment]
Fair enough :)

**Pull request merged**